metadata-ingestion/src/datahub/utilities/sql_lineage_parser_impl.py

```diff
@@ -32,18 +35,59 @@ def __init__(self, sql_query: str) -> None:
         if "lateral flatten" in sql_query:
             sql_query = sql_query[: sql_query.find("lateral flatten")]
 
+        # BigQuery can use # as a comment symbol and that may break the parsing
+        # if the comment starts without a space sign after #
+        # and the comment is written inside SQL DDL statement (e.g., after CREATE...AS)
+        sql_query = re.sub(r"#([^ ])", r"# \1", sql_query, flags=re.IGNORECASE)
+
+        # Wrap calls to field '<table_alias>.from' in ` so it would not be taken as a reserved keyword
+        sql_query = re.sub(r"(\w*\.from)", r"`\1`", sql_query, flags=re.IGNORECASE)
+
+        # Apply sqlparser formatting to get rid of comments and reindent keywords
+        # which should remove some potential inconsistencies in parsing output
+        sql_query = sqlparse.format(
+            sql_query.strip(),
+            reindent_aligned=True,
+            strip_comments=True,
+        )
+
+        # SqlLineageParser does not allow table/view names not being wrapped in quotes or backticks
+        # Add ` signs before and after supposed object name that comes right after reserved word FROM
+        # note 1: this excludes date/time/datetime extract functions like EXTRACT DATE FROM...
+        # note 2: this includes adding ` signs to CTE aliases
+        sql_query = re.sub(
+            r"(?<!day\s)(?<!(date|time|hour|week|year)\s)(?<!month\s)(?<!(second|minute)\s)(?<!quarter\s)(?<!\.)(from\s)([^`\s()]+)",
```

Comment:
Can you please add examples for this regex? Please extract the regex part in a separate functions and add unit tests for what the regex is supposed to be doing.

Reply:
Wrapped into its own function and added unit tests